GNUmakefile

```diff
@@ -1,18 +1,31 @@
 TEST?=$$(go list ./... |grep -v 'vendor')
 GOFMT_FILES?=$$(find . -name '*.go' |grep -v vendor)
 
+ifndef TF_PARA_NUM
+	TF_PARA_NUM=20
+endif
+
 default: build
 
 build: fmtcheck
 	go install
 
-test: fmtcheck
+test-install:
 	go test -i $(TEST) || exit 1
+
+test: fmtcheck test-install
 	echo $(TEST) | \
 		xargs -t -n4 go test $(TESTARGS) -timeout=30s -parallel=4
 
-testacc: fmtcheck
-	TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 180m
+# example usage:
+#  1. Enable parallel run for all acceptance tests with default number(20) of
+#  threads:
+#    TF_PARA=1 make testacc
+#  2. Enable parallel run for acceptance tests with matching pattern.
+#    TF_PARA=1 TESTARGS="-run TestAccAzureRMStorage" make testacc
+testacc: fmtcheck test-install
```

Comment:
I am not the original author of test/testacc here. From their original definitions:
test: fmtcheck, test-install, unit-test
testacc: fmtcheck, acc-test.

What I changed here is just add an extra step, test-install, into testacc. Do you want me to include the unit test into acc test ? I checked the resource.Test, it will run unit tests by default, but will skip acc tests if TF_ACC is not set.

Reply:
> What I changed here is just add an extra step, test-install, into testacc.

I was thinking more of removing `test-install` from `testacc` - as it's not needed since we should be running `make test` first anyway (which will catch this). The benefit to having these split out is it's easier to expose what's failed (i.e. unit/acc tests), rather than just "some tests failed" - if that makes sense?

```
make test
make acctest
```

> Do you want me to include the unit test into acc test ? I checked the resource.Test, it will run unit tests by default, but will skip acc tests if TF_ACC is not set.

No that's fine - if we can just remove `test-install` this should be fine :)